src/clj/game/cards/ice.clj

```diff
@@ -3697,6 +3697,7 @@
                              :cancel-effect (effect (system-msg "uses Wave to shuffle R&D")
                                                     (shuffle! :deck))
                              :effect (effect (shuffle! :deck)
+                                             (reveal eid target)
```

Comment:
Gotta be a `wait-for`, add `:async true` to the ability map, and then add `effect-completed` after `move`